### PR TITLE
Apply discount

### PR DESCRIPTION
**Pull request by kprakobkit**

### CODE REVIEW

File: models/order.js [hunk 1]
```diff
@@ -130,6 +144,32 @@ function populateTransaction(order) {
   });
 }
 
+function getDiscounts(orders) {
+  return Promise.all(orders.map(populateDiscount));
+}
+
+function populateDiscount(order) {
```
Comment (achiu8) [inline models/order.js:151]: Yeah this model is getting ridiculous haha..

File: src/components/ApplyDiscount.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    this.state = {
+      discounts: this.props.appliedDiscounts
+    };
+  }
+
+  handleAddDiscount(discount) {
+    const { _id } = discount;
+    const idx = _.findIndex(_.propEq('_id', _id))(this.state.discounts);
+    const updatedDiscounts = idx > -1 ?
+      _.remove(idx, 1)(this.state.discounts) :
```
Comment (achiu8) [inline src/components/ApplyDiscount.js:20]: Why do we remove an existing discount without adding the new one?

File: src/components/ProcessingOrder.js [hunk 3]
```diff
@@ -22,11 +31,20 @@ class ProcessingOrder extends Component {
   }
 
   tax() {
-    return Math.round(this.subtotal() * constants.TAX_RATE);
+    return Math.round((this.subtotal() - this.discount()) * constants.TAX_RATE);
+  }
+
+  discount() {
```
Comment (achiu8) [inline src/components/ProcessingOrder.js:37]: Could we just simplify this to the below?

```
const percentage = _.sum(this.props.order.discounts);
return this.subtotal() * percentage;
```

File: src/components/ProcessingOrder.js [hunk 4]
```diff
@@ -62,12 +89,28 @@ class ProcessingOrder extends Component {
           { className: 'table table-striped' },
           dom.tbody(
             null,
+            dom.tr(
+              null,
+              dom.th(null, 'Item'),
+              dom.th(null, 'Quantity'),
+              dom.th({ className: 'text-right' }, status === constants.READY_FOR_BILL ? dom.button({ className: 'btn btn-default apply-discount', onClick: this.openModal }, 'Discount') : null)
```
Comment (achiu8) [inline src/components/ProcessingOrder.js]: Can we wrap this?

File: src/components/ApplyDiscount.js (hunk 2 above)
Comment (kprakobkit) [inline src/components/ApplyDiscount.js:20]: Removing it to toggle selecting/deselecting, and am using state.discounts to display what's selected. does that make sense? maybe the naming is confusing...?

**Comment by kprakobkit** [comment]
found a bug closing for now.

File: src/components/ApplyDiscount.js (hunk 2 above)
Comment (achiu8) [inline src/components/ApplyDiscount.js:20]: Ah, I see now..

**Pull request closed without merging**